### ISSUE TITLE
Link fails when runtime built with -g

### ISSUE THREAD

**Issue by jclark** [ballerina-platform/nballerina#957]
If the runtime directory is built with `make CFLAGS=-g`, then in the `test` directory `make 14-bug` gets the error:

```
llvm-link-12 -o result/floatsubtypecompare-v.bc ll/floatsubtypecompare-v.ll ../../../runtime/balrt_inline.bc
inlinable function call in a function with debug info must have a !dbg location
  %27 = call i8 addrspace(1)* @_bal_float_to_tagged(double 1.000000e+00)
inlinable function call in a function with debug info must have a !dbg location
  %29 = call i8 addrspace(1)* @_bal_float_to_tagged(double 2.000000e+00)
inlinable function call in a function with debug info must have a !dbg location
  %31 = call i8 addrspace(1)* @_bal_float_to_tagged(double 3.000000e+00)
llvm-link-12: error: linked module is broken!
```

**Comment by heshanpadmasiri** [comment]
@jclark based on my testing ([demo repo](https://github.com/heshanpadmasiri/llvm-link-debug)) 3 conditions that must be true for the error to occur,

1. runtime (`.bc` file) is compile with `-g` flag on (I managed to produce the error without both static and inline explicitly stated but with -O2 in the [header file](https://github.com/heshanpadmasiri/llvm-link-debug/blob/7df19009735b4154542ca830730ce1781f72e112/runtime.h), ie. compile with `make CFLAGS=-g`)

2. `ll` file that call the runtime function has a debug location ([line](https://github.com/heshanpadmasiri/llvm-link-debug/blob/7df19009735b4154542ca830730ce1781f72e112/main.ll#L3)) (removing this also remove the error)

3. call instruction don't have a debug location (adding a debug location [here](https://github.com/heshanpadmasiri/llvm-link-debug/blob/7df19009735b4154542ca830730ce1781f72e112/main.ll#L13) also fix the issue)

_1 may not be true for any function since error still says `inlinable`_

It also seems based on the currently committed ll [files](https://github.com/ballerina-platform/nballerina/blob/4dd9130feb891cf1cfad3ddb692737155a05f953/test/ll/06-float/02-v.ll#L23) 3 didn't used to be true

**Comment by jclark** [comment]
So the fix is to add the debug location to the call instruction?

Can you find the commit where calls stopped having the debug location?

**Comment by heshanpadmasiri** [comment]
> So the fix is to add the debug location to the call instruction?
> 
> Can you find the commit where calls stopped having the debug location?

Yes. I'll make a pr for that.
It seems that first time llvm-link complained about this issue was with bf3618503fd88f1757c9076e50f7d4e541121e41. But that fixed another issue (#636) where llvm-link was complaining about another problem when runtime is complied with -g. So we may had this even before that and #636 was hiding it.

**Comment by heshanpadmasiri** [comment]
There seems to be another bug when we compile runtime with `-g`. It causes llvm to segfault giving an error message as fallows in few test cases (such as [`02-misc/stackoverflow-p.bal`](https://github.com/ballerina-platform/nballerina/blob/main/compiler/testSuite/02-misc/stackoverflow-p.bal)):
```
PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:
0.	Program arguments: /usr/lib/llvm-12/bin/clang -cc1 -triple x86_64-pc-linux-gnu -emit-obj --mrelax-relocations -disable-free -disable-llvm-verifier -discard-value-names -main-file-name 1-v.bc -mrelocation-model static -mframe-pointer=none -fmath-errno -fno-rounding-math -mconstructor-aliases -munwind-tables -target-cpu x86-64 -tune-cpu generic -fno-split-dwarf-inlining -debug-info-kind=limited -dwarf-version=4 -debugger-tuning=gdb -resource-dir /usr/lib/llvm-12/lib/clang/12.0.1 -O2 -fdebug-compilation-dir /home/heshan/work/nballerina/test/out/04-typecast -ferror-limit 19 -fgnuc-version=4.2.1 -vectorize-loops -vectorize-slp -faddrsig -o /tmp/1-v-b8a31e.o -x ir result/1-v.bc
1.	Code generation
PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:
0.	Program arguments: /usr/lib/llvm-12/bin/clang -cc1 -triple x86_64-pc-linux-gnu -emit-obj --mrelax-relocations -disable-free -disable-llvm-verifier -discard-value-names -main-file-name 2-v.bc -mrelocation-model static -mframe-pointer=none -fmath-errno -fno-rounding-math -mconstructor-aliases -munwind-tables -target-cpu x86-64 -tune-cpu generic -fno-split-dwarf-inlining -debug-info-kind=limited -dwarf-version=4 -debugger-tuning=gdb -resource-dir /usr/lib/llvm-12/lib/clang/12.0.1 -O2 -fdebug-compilation-dir /home/heshan/work/nballerina/test/out/04-typecast -ferror-limit 19 -fgnuc-version=4.2.1 -vectorize-loops -vectorize-slp -faddrsig -o /tmp/2-v-e8462f.o -x ir result/2-v.bc
1.	Code generation
 #0 0x00007febb6c76871 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0xd26871)
 #1 0x00007febb6c749a0 llvm::sys::RunSignalHandlers() (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0xd249a0)
 #2 0x00007febb6c76f82 (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0xd26f82)
 #3 0x00007febbe7c13c0 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x143c0)
 #4 0x00007febb7412700 llvm::DIE::getUnitDie() const (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0x14c2700)
 #5 0x00007febb742646f llvm::DwarfDebug::finalizeModuleInfo() (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0x14d646f)
 #6 0x00007febb7426b38 llvm::DwarfDebug::endModule() (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0x14d6b38)
 #7 0x00007febb73fc191 llvm::AsmPrinter::doFinalization(llvm::Module&) (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0x14ac191)
 #8 0x00007febb6db0161 llvm::FPPassManager::doFinalization(llvm::Module&) (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0xe60161)
 #9 0x00007febb6daabf2 llvm::legacy::PassManagerImpl::run(llvm::Module&) (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0xe5abf2)
#10 0x00007febbd014ac6 clang::EmitBackendOutput(clang::DiagnosticsEngine&, clang::HeaderSearchOptions const&, clang::CodeGenOptions const&, clang::TargetOptions const&, clang::LangOptions const&, llvm::DataLayout const&, llvm::Module*, clang::BackendAction, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream> >) (/lib/x86_64-linux-gnu/libclang-cpp.so.12+0x15a9ac6)
#11 0x00007febbd2beffc clang::CodeGenAction::ExecuteAction() (/lib/x86_64-linux-gnu/libclang-cpp.so.12+0x1853ffc)
#12 0x00007febbd9f8456 clang::FrontendAction::Execute() (/lib/x86_64-linux-gnu/libclang-cpp.so.12+0x1f8d456)
#13 0x00007febbd980331 clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/lib/x86_64-linux-gnu/libclang-cpp.so.12+0x1f15331)
#14 0x00007febbda5f8e6 clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/lib/x86_64-linux-gnu/libclang-cpp.so.12+0x1ff48e6)
#15 0x0000000000412ab9 cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (/usr/lib/llvm-12/bin/clang+0x412ab9)
#16 0x0000000000410dbf (/usr/lib/llvm-12/bin/clang+0x410dbf)
#17 0x0000000000410bb6 main (/usr/lib/llvm-12/bin/clang+0x410bb6)
#18 0x00007febb5a340b3 __libc_start_main /build/glibc-sMfBJT/glibc-2.31/csu/../csu/libc-start.c:342:3
#19 0x000000000040ddce _start (/usr/lib/llvm-12/bin/clang+0x40ddce)
 #0 0x00007f132ae59871 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0xd26871)
 #1 0x00007f132ae579a0 llvm::sys::RunSignalHandlers() (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0xd249a0)
 #2 0x00007f132ae59f82 (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0xd26f82)
 #3 0x00007f13329a43c0 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x143c0)
 #4 0x00007f132b5f5700 llvm::DIE::getUnitDie() const (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0x14c2700)
 #5 0x00007f132b60946f llvm::DwarfDebug::finalizeModuleInfo() (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0x14d646f)
 #6 0x00007f132b609b38 llvm::DwarfDebug::endModule() (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0x14d6b38)
 #7 0x00007f132b5df191 llvm::AsmPrinter::doFinalization(llvm::Module&) (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0x14ac191)
 #8 0x00007f132af93161 llvm::FPPassManager::doFinalization(llvm::Module&) (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0xe60161)
 #9 0x00007f132af8dbf2 llvm::legacy::PassManagerImpl::run(llvm::Module&) (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0xe5abf2)
#10 0x00007f13311f7ac6 clang::EmitBackendOutput(clang::DiagnosticsEngine&, clang::HeaderSearchOptions const&, clang::CodeGenOptions const&, clang::TargetOptions const&, clang::LangOptions const&, llvm::DataLayout const&, llvm::Module*, clang::BackendAction, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream> >) (/lib/x86_64-linux-gnu/libclang-cpp.so.12+0x15a9ac6)
#11 0x00007f13314a1ffc clang::CodeGenAction::ExecuteAction() (/lib/x86_64-linux-gnu/libclang-cpp.so.12+0x1853ffc)
#12 0x00007f1331bdb456 clang::FrontendAction::Execute() (/lib/x86_64-linux-gnu/libclang-cpp.so.12+0x1f8d456)
#13 0x00007f1331b63331 clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/lib/x86_64-linux-gnu/libclang-cpp.so.12+0x1f15331)
#14 0x00007f1331c428e6 clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/lib/x86_64-linux-gnu/libclang-cpp.so.12+0x1ff48e6)
#15 0x0000000000412ab9 cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (/usr/lib/llvm-12/bin/clang+0x412ab9)
#16 0x0000000000410dbf (/usr/lib/llvm-12/bin/clang+0x410dbf)
#17 0x0000000000410bb6 main (/usr/lib/llvm-12/bin/clang+0x410bb6)
#18 0x00007f1329c170b3 __libc_start_main /build/glibc-sMfBJT/glibc-2.31/csu/../csu/libc-start.c:342:3
#19 0x000000000040ddce _start (/usr/lib/llvm-12/bin/clang+0x40ddce)
clang: error: unable to execute command: Segmentation fault (core dumped)
clang: error: unable to execute command: Segmentation fault (core dumped)
clang: error: clang frontend command failed due to signal (use -v to see invocation)
Ubuntu clang version 12.0.1-++20211029101322+fed41342a82f-1~exp1~20211029221816.4
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
clang: error: clang frontend command failed due to signal (use -v to see invocation)
Ubuntu clang version 12.0.1-++20211029101322+fed41342a82f-1~exp1~20211029221816.4
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

**Comment by heshanpadmasiri** [comment]
@jclark in order to simplify our current approach towards setting and clearing the debug location I think we should separate the setting debug location `position` and `origin` (currently we do both [here](https://github.com/ballerina-platform/nballerina/blob/16c4dee3660f681b38cd8283228b77c464574e07/compiler/modules/nback/scaffold.bal#L290)) in the scaffold
1. `function setDebugLocation(llvm:Builder builder, bir:Position pos)`
  In the case of full debug setting debug location is passed  to the `llvm:Builder`. Otherwise it set the debug location in scaffold     but don’t pass it to llvm.
  This would be called at the beging of every bir instruction [here](https://github.com/ballerina-platform/nballerina/blob/16c4dee3660f681b38cd8283228b77c464574e07/compiler/modules/nback/basic.bal#L50) and at the beginning of the prolog [here](https://github.com/ballerina-platform/nballerina/blob/16c4dee3660f681b38cd8283228b77c464574e07/compiler/modules/nback/basic.bal#L34)

2. `function setDebugOrigin(DebugLocationOrigin origin)`
  This will use the same rules to decide whether or not to send the debug location to llvm as current [`setDebugLocation`](https://github.com/ballerina-platform/nballerina/blob/16c4dee3660f681b38cd8283228b77c464574e07/compiler/modules/nback/scaffold.bal#L290)
So anywhere we need to send the debug location specifically to llvm (such as function calls) we will call this with the correct origin type.
They can revert to default behaviour by setting this to `DEBUG_ORIGIN_OTHER` in places like [here](https://github.com/ballerina-platform/nballerina/blob/16c4dee3660f681b38cd8283228b77c464574e07/compiler/modules/nback/build.bal#L159) (basically this function will clear the debug location in the builder if under current rules we don't send the debug location to llvm)

With the above change i think we no longer need `clearDebugLocation` since 
1. Each new instruction will update the debug location overriding the previous
2. We can use `setDebugOrigin` to clear debug location in llvm in cases where we set the debug location only to a part of the instruction (if not full debug)

Main problem I see in this approach is this will make scaffold stateful (with regard to DebugLocation) but this will allow us the set the debug location in only one place and only change how it is passed to llvm in other places

To simplify the repeated runtime function calls I'll add the fallowing function to scaffold
```
function callRuntimeFunction(llvm:Builder builder,  RuntimeFunction rf, llvm:Value[] args) returns llvm:Value?
```
(This will use the the new `setDebugOrigin` function to pass the debug location to llvm.)

**Comment by heshanpadmasiri** [comment]
> There seems to be another bug when we compile runtime with `-g`. It causes llvm to segfault giving an error message as fallows in few test cases (such as [`02-misc/stackoverflow-p.bal`](https://github.com/ballerina-platform/nballerina/blob/main/compiler/testSuite/02-misc/stackoverflow-p.bal)):
> 
> ```
> PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace, preprocessed source, and associated run script.
> Stack dump:
> 0.	Program arguments: /usr/lib/llvm-12/bin/clang -cc1 -triple x86_64-pc-linux-gnu -emit-obj --mrelax-relocations -disable-free -disable-llvm-verifier -discard-value-names -main-file-name 1-v.bc -mrelocation-model static -mframe-pointer=none -fmath-errno -fno-rounding-math -mconstructor-aliases -munwind-tables -target-cpu x86-64 -tune-cpu generic -fno-split-dwarf-inlining -debug-info-kind=limited -dwarf-version=4 -debugger-tuning=gdb -resource-dir /usr/lib/llvm-12/lib/clang/12.0.1 -O2 -fdebug-compilation-dir /home/heshan/work/nballerina/test/out/04-typecast -ferror-limit 19 -fgnuc-version=4.2.1 -vectorize-loops -vectorize-slp -faddrsig -o /tmp/1-v-b8a31e.o -x ir result/1-v.bc
> 1.	Code generation
> PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace, preprocessed source, and associated run script.
> Stack dump:
> 0.	Program arguments: /usr/lib/llvm-12/bin/clang -cc1 -triple x86_64-pc-linux-gnu -emit-obj --mrelax-relocations -disable-free -disable-llvm-verifier -discard-value-names -main-file-name 2-v.bc -mrelocation-model static -mframe-pointer=none -fmath-errno -fno-rounding-math -mconstructor-aliases -munwind-tables -target-cpu x86-64 -tune-cpu generic -fno-split-dwarf-inlining -debug-info-kind=limited -dwarf-version=4 -debugger-tuning=gdb -resource-dir /usr/lib/llvm-12/lib/clang/12.0.1 -O2 -fdebug-compilation-dir /home/heshan/work/nballerina/test/out/04-typecast -ferror-limit 19 -fgnuc-version=4.2.1 -vectorize-loops -vectorize-slp -faddrsig -o /tmp/2-v-e8462f.o -x ir result/2-v.bc
> 1.	Code generation
>  #0 0x00007febb6c76871 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0xd26871)
>  #1 0x00007febb6c749a0 llvm::sys::RunSignalHandlers() (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0xd249a0)
>  #2 0x00007febb6c76f82 (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0xd26f82)
>  #3 0x00007febbe7c13c0 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x143c0)
>  #4 0x00007febb7412700 llvm::DIE::getUnitDie() const (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0x14c2700)
>  #5 0x00007febb742646f llvm::DwarfDebug::finalizeModuleInfo() (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0x14d646f)
>  #6 0x00007febb7426b38 llvm::DwarfDebug::endModule() (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0x14d6b38)
>  #7 0x00007febb73fc191 llvm::AsmPrinter::doFinalization(llvm::Module&) (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0x14ac191)
>  #8 0x00007febb6db0161 llvm::FPPassManager::doFinalization(llvm::Module&) (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0xe60161)
>  #9 0x00007febb6daabf2 llvm::legacy::PassManagerImpl::run(llvm::Module&) (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0xe5abf2)
> #10 0x00007febbd014ac6 clang::EmitBackendOutput(clang::DiagnosticsEngine&, clang::HeaderSearchOptions const&, clang::CodeGenOptions const&, clang::TargetOptions const&, clang::LangOptions const&, llvm::DataLayout const&, llvm::Module*, clang::BackendAction, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream> >) (/lib/x86_64-linux-gnu/libclang-cpp.so.12+0x15a9ac6)
> #11 0x00007febbd2beffc clang::CodeGenAction::ExecuteAction() (/lib/x86_64-linux-gnu/libclang-cpp.so.12+0x1853ffc)
> #12 0x00007febbd9f8456 clang::FrontendAction::Execute() (/lib/x86_64-linux-gnu/libclang-cpp.so.12+0x1f8d456)
> #13 0x00007febbd980331 clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/lib/x86_64-linux-gnu/libclang-cpp.so.12+0x1f15331)
> #14 0x00007febbda5f8e6 clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/lib/x86_64-linux-gnu/libclang-cpp.so.12+0x1ff48e6)
> #15 0x0000000000412ab9 cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (/usr/lib/llvm-12/bin/clang+0x412ab9)
> #16 0x0000000000410dbf (/usr/lib/llvm-12/bin/clang+0x410dbf)
> #17 0x0000000000410bb6 main (/usr/lib/llvm-12/bin/clang+0x410bb6)
> #18 0x00007febb5a340b3 __libc_start_main /build/glibc-sMfBJT/glibc-2.31/csu/../csu/libc-start.c:342:3
> #19 0x000000000040ddce _start (/usr/lib/llvm-12/bin/clang+0x40ddce)
>  #0 0x00007f132ae59871 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0xd26871)
>  #1 0x00007f132ae579a0 llvm::sys::RunSignalHandlers() (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0xd249a0)
>  #2 0x00007f132ae59f82 (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0xd26f82)
>  #3 0x00007f13329a43c0 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x143c0)
>  #4 0x00007f132b5f5700 llvm::DIE::getUnitDie() const (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0x14c2700)
>  #5 0x00007f132b60946f llvm::DwarfDebug::finalizeModuleInfo() (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0x14d646f)
>  #6 0x00007f132b609b38 llvm::DwarfDebug::endModule() (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0x14d6b38)
>  #7 0x00007f132b5df191 llvm::AsmPrinter::doFinalization(llvm::Module&) (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0x14ac191)
>  #8 0x00007f132af93161 llvm::FPPassManager::doFinalization(llvm::Module&) (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0xe60161)
>  #9 0x00007f132af8dbf2 llvm::legacy::PassManagerImpl::run(llvm::Module&) (/lib/x86_64-linux-gnu/libLLVM-12.so.1+0xe5abf2)
> #10 0x00007f13311f7ac6 clang::EmitBackendOutput(clang::DiagnosticsEngine&, clang::HeaderSearchOptions const&, clang::CodeGenOptions const&, clang::TargetOptions const&, clang::LangOptions const&, llvm::DataLayout const&, llvm::Module*, clang::BackendAction, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream> >) (/lib/x86_64-linux-gnu/libclang-cpp.so.12+0x15a9ac6)
> #11 0x00007f13314a1ffc clang::CodeGenAction::ExecuteAction() (/lib/x86_64-linux-gnu/libclang-cpp.so.12+0x1853ffc)
> #12 0x00007f1331bdb456 clang::FrontendAction::Execute() (/lib/x86_64-linux-gnu/libclang-cpp.so.12+0x1f8d456)
> #13 0x00007f1331b63331 clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/lib/x86_64-linux-gnu/libclang-cpp.so.12+0x1f15331)
> #14 0x00007f1331c428e6 clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/lib/x86_64-linux-gnu/libclang-cpp.so.12+0x1ff48e6)
> #15 0x0000000000412ab9 cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (/usr/lib/llvm-12/bin/clang+0x412ab9)
> #16 0x0000000000410dbf (/usr/lib/llvm-12/bin/clang+0x410dbf)
> #17 0x0000000000410bb6 main (/usr/lib/llvm-12/bin/clang+0x410bb6)
> #18 0x00007f1329c170b3 __libc_start_main /build/glibc-sMfBJT/glibc-2.31/csu/../csu/libc-start.c:342:3
> #19 0x000000000040ddce _start (/usr/lib/llvm-12/bin/clang+0x40ddce)
> clang: error: unable to execute command: Segmentation fault (core dumped)
> clang: error: unable to execute command: Segmentation fault (core dumped)
> clang: error: clang frontend command failed due to signal (use -v to see invocation)
> Ubuntu clang version 12.0.1-++20211029101322+fed41342a82f-1~exp1~20211029221816.4
> Target: x86_64-pc-linux-gnu
> Thread model: posix
> InstalledDir: /usr/bin
> clang: error: clang frontend command failed due to signal (use -v to see invocation)
> Ubuntu clang version 12.0.1-++20211029101322+fed41342a82f-1~exp1~20211029221816.4
> Target: x86_64-pc-linux-gnu
> Thread model: posix
> InstalledDir: /usr/bin
> ```

This is also related to the positions (a position that linker did not complain about but adding it fixed the problem)  will be fixed by #964

**Comment by jclark** [comment]
It doesn't make sense to me to continue to call this DebugLocationOrigin, nor to call the method setDebugOrigin. This is just not what the method is doing.  Maybe DebugLocationUsage and useDebugLocation?

More consistent name for function would be buildRuntimeFunctionCall.
